internal/macaroon/expirablestorage.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package macaroon
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery"
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery/dbrootkeystore"
+	"github.com/juju/errors"
+)
+
+// Functions defined as variables so they can be overridden
+// for testing.
+var (
+	Clock dbrootkeystore.Clock
+)
```

Comment:
Pass the Clock a as an optional dependency to the `NewExpirableStorage` . That way we don't need to patch the function out during tests.

Reply:
Yeah agree.

However, making a clock mandatory turned out to be by far the cleanest way to do this.